distribution/bwc/build.gradle

```diff
@@ -17,233 +17,224 @@
  * under the License.
  */
 
-
-
 import org.apache.tools.ant.taskdefs.condition.Os
 import org.elasticsearch.gradle.LoggedExec
 import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.VersionCollection
 
 import java.nio.charset.StandardCharsets
 
 import static org.elasticsearch.gradle.BuildPlugin.getJavaHome
+
 /**
- * This is a dummy project which does a local checkout of the previous
- * wire compat version's branch, and builds a snapshot. This allows backcompat
- * tests to test against the next unreleased version, closest to this version,
- * without relying on snapshots.
+ *  We want to be able to do BWC tests for unreleased versions without relying on and waiting for snapshots.
+ *  For this we need to check out and build the unreleased versions.
+ *  Since These depend on the current version, we can't name the Gradle projects statically, and don't know what the
+ *  unreleased versions are when Gradle projects are set up, so we use "build-unreleased-version-*" as placeholders
+ *  and configure them to build various versions here.
  */
-subprojects {
-
-  Version bwcVersion = bwcVersions.getSnapshotForProject(project.name)
-  if (bwcVersion == null) {
-    // this project wont do anything
-    return
-  }
-
-  String bwcBranch
-  if (project.name == 'next-minor-snapshot') {
-    // this is always a .x series
-    bwcBranch = "${bwcVersion.major}.x"
-  } else {
-    bwcBranch = "${bwcVersion.major}.${bwcVersion.minor}"
-  }
-
-  apply plugin: 'distribution'
-  // Not published so no need to assemble
-  assemble.enabled = false
-  assemble.dependsOn.remove('buildBwcVersion')
-
-  File checkoutDir = file("${buildDir}/bwc/checkout-${bwcBranch}")
-
-  final String remote = System.getProperty("tests.bwc.remote", "elastic")
-
-  final boolean gitFetchLatest
-  final String gitFetchLatestProperty = System.getProperty("tests.bwc.git_fetch_latest", "true")
-  if ("true".equals(gitFetchLatestProperty)) {
-    gitFetchLatest = true
-  } else if ("false".equals(gitFetchLatestProperty)) {
-    gitFetchLatest = false
-  } else {
-    throw new GradleException("tests.bwc.git_fetch_latest must be [true] or [false] but was [" + gitFetchLatestProperty + "]")
-  }
-
-  task createClone(type: LoggedExec) {
-    onlyIf { checkoutDir.exists() == false }
-    commandLine = ['git', 'clone', rootDir, checkoutDir]
-  }
-
-  task findRemote(type: LoggedExec) {
-    dependsOn createClone
-    workingDir = checkoutDir
-    commandLine = ['git', 'remote', '-v']
-    ByteArrayOutputStream output = new ByteArrayOutputStream()
-    standardOutput = output
-    doLast {
-      project.ext.remoteExists = false
-      output.toString('UTF-8').eachLine {
-        if (it.contains("${remote}\t")) {
-          project.ext.remoteExists = true
+bwcVersions.forPreviousUnreleased { VersionCollection.UnreleasedVersionDescription unreleasedVersion ->
+    project("$project.path:$unreleasedVersion.gradleProjectName") {
```

Comment:
Can you please use `${}` style substitutions, so it is obvious where the sub ends and the literal string begins?